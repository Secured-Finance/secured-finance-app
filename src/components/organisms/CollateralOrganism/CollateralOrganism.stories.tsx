import { ComponentMeta, ComponentStory } from '@storybook/react';
import { withWalletProvider } from 'src/../.storybook/decorators';
import { CollateralOrganism } from './CollateralOrganism';

export default {
    title: 'Organism/CollateralOrganism',
    component: CollateralOrganism,
    args: {},
    decorators: [withWalletProvider],
} as ComponentMeta<typeof CollateralOrganism>;

const Template: ComponentStory<typeof CollateralOrganism> = () => (
    <CollateralOrganism />
);

export const Default = Template.bind({});
