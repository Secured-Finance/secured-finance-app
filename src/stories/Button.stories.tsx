import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Button, IButton } from 'src/components/new/Button';

export default {
    title: 'Components/Button',
    component: Button,
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

const Template: Story<IButton> = args => <Button {...args}>Button</Button>;

export const Default = Template.bind({});
Default.args = {
    outline: false,
};

export const Outlined = Template.bind({});
Outlined.args = {
    outline: true,
};
