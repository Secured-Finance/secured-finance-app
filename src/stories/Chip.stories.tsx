import React from 'react';
import { Story, Meta } from '@storybook/react';
import cm from './index.module.scss';

import { ChipButton, IChipButton } from 'src/components/new/Chip/ChipButton';
import { ChipLabel, IChipLabel } from 'src/components/new/Chip/ChipLabel';

export default {
    title: 'Components/Chips',
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

export const ChipButtons: Story<IChipButton> = () => (
    <div className={cm.elementsContainer}>
        <ChipButton>Button</ChipButton>
        <ChipButton accent={'purple'}>Button</ChipButton>
        <ChipButton accent={'red'}>Button</ChipButton>
    </div>
);

export const ChipLabels: Story<IChipLabel> = () => (
    <div className={cm.elementsContainer}>
        <ChipLabel label={'Registered'} />
        <ChipLabel label={'Working'} accent={'green'} />
        <ChipLabel label={'Terminated'} accent={'red'} />
        <ChipLabel label={'Closed'} accent={'grey'} />
        <ChipLabel label={'Past Due'} accent={'turquoise'} />
        <ChipLabel label={'Due'} accent={'blue'} />
    </div>
);
