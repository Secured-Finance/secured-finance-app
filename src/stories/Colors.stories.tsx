import { Meta, Story } from '@storybook/react';

export default {
    title: 'Design System/Colors',
    parameters: {
        chromatic: { disableSnapshot: false },
    },
} as Meta;

export const Primary: Story = () => (
    <div className='inline-flex flex-col items-center justify-start rounded-3xl bg-white p-10'>
        <div className='flex flex-col items-start justify-start space-y-6'>
            <p className='text-lg font-semibold leading-10 text-black-80'>
                Primary
            </p>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-universeBlue' />
                <div className='h-10 w-10 rounded-lg bg-horizonBlue' />
                <div className='h-10 w-10 rounded-lg bg-starBlue' />
                <div className='h-10 w-10 rounded-lg bg-teal' />
                <div className='h-10 w-10 rounded-lg bg-green' />
                <div className='h-10 w-10 rounded-lg bg-orange' />
                <div className='h-10 w-10 rounded-lg bg-red' />
            </div>
            <div className='h-10 w-10 rounded-lg bg-purple' />
        </div>
    </div>
);

export const Secondary: Story = () => (
    <div className='inline-flex flex-col items-start justify-start space-y-6 rounded-3xl bg-white p-10'>
        <p className='text-lg font-semibold leading-10 text-black-80'>
            Secondary
        </p>
        <div className='inline-flex items-center justify-start space-x-6'>
            <div className='h-10 w-10 rounded-lg bg-galacticOrange' />
            <div className='h-10 w-10 rounded-lg bg-planetaryPurple' />
            <div className='h-10 w-10 rounded-lg bg-nebulaTeal' />
            <div className='h-10 w-10 rounded-lg bg-secondaryPurple ' />
            <div className='h-10 w-10 rounded-lg bg-moonGrey' />
        </div>
    </div>
);

export const Shades: Story = () => (
    <div className='inline-flex flex-col items-center justify-start rounded-3xl bg-white p-10'>
        <div className='flex flex-col items-start justify-start space-y-6'>
            <p className='text-lg font-semibold leading-10 text-black-80'>
                Shades
            </p>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-starBlue-80' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-60' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-40' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-20' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-10' />
            </div>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-black-90' />
                <div className='h-10 w-10 rounded-lg bg-black-80' />
                <div className='h-10 w-10 rounded-lg bg-black-70' />
                <div className='h-10 w-10 rounded-lg bg-black-60' />
                <div className='h-10 w-10 rounded-lg bg-black-50' />
                <div className='h-10 w-10 rounded-lg bg-black-40' />
                <div className='h-10 w-10 rounded-lg bg-black-30' />
                <div className='h-10 w-10 rounded-lg bg-black-20' />
                <div className='h-10 w-10 rounded-lg bg-black-10' />
            </div>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg border-2 bg-white-90' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-80' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-70' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-60' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-50' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-40' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-30' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-20' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-10' />
            </div>
        </div>
    </div>
);
