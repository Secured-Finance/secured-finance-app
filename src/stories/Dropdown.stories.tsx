import React from 'react';
import { Story, Meta } from '@storybook/react';
import { Dropdown, IDropdown } from 'src/components/new/Dropdown';
import { filecoin, telegram } from 'src/components/new/icons';

export default {
    title: 'Components/Dropdown',
    component: Dropdown,
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

const Template: Story<IDropdown> = args => {
    const [value, setValue] = React.useState(args.value ?? '');

    return (
        <Dropdown
            {...args}
            onChange={e => {
                setValue(e.target.value);
            }}
            value={value}
        />
    );
};

let value = 1;
export const Default = Template.bind({});
Default.args = {
    options: [
        { value: 1, label: 'One' },
        { value: 2, label: 'Two', icon: filecoin },
        { value: 3, label: 'Three looooong' },
        { value: 4, label: 'Long with icon', icon: telegram },
    ],
    value,
    label: '',
    noBorders: false,
};
