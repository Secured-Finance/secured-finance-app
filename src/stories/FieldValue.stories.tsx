import React from 'react';
import { Story, Meta } from '@storybook/react';
import { FieldValue, IFieldValue } from 'src/components/new/FieldValue';

export default {
    title: 'Components/FieldValue',
    component: FieldValue,
} as Meta;

const Template: Story<IFieldValue> = args => <FieldValue {...args} />;

export const Default = Template.bind({});
Default.args = {
    field: '1Y Yield',
    value: '4.25%',
    small: false,
    bold: false,
};

export const Small = Template.bind({});
Small.args = {
    field: '1Y Yield',
    value: '4.25%',
    small: true,
};

export const Bold = Template.bind({});
Bold.args = {
    field: '1Y Yield',
    value: '4.25%',
    bold: true,
};
