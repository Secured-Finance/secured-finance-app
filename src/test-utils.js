import { render as rtlRender } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from 'src/store/reducers';

function render(
    ui,
    {
        initialState,
        store = createStore(reducer, initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>;
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
