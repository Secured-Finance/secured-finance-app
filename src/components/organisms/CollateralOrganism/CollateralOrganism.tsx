import { Tab } from 'src/components/molecules';
import { CollateralTab } from 'src/components/organisms';
import { useCollateralBook } from 'src/hooks';
import { useWallet } from 'use-wallet';

const tabDataArray = [{ text: 'Asset Management' }];

export const CollateralOrganism = () => {
    const { account } = useWallet();
    const collateralBook = useCollateralBook(account);

    return (
        <div className='h-fit w-full'>
            <Tab tabDataArray={tabDataArray}>
                <CollateralTab collateralBook={collateralBook} />
            </Tab>
        </div>
    );
};
