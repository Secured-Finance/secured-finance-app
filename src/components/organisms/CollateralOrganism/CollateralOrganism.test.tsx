import { composeStories } from '@storybook/testing-react';
import { queries, render, RenderResult, waitFor } from 'src/test-utils.js';
import * as stories from './CollateralOrganism.stories';

const { Default } = composeStories(stories);

describe('CollateralOrganism Component', () => {
    it('should render CollateralOrganism', async () => {
        let ag: RenderResult<typeof queries, HTMLElement>;
        await waitFor(() => {
            ag = render(<Default />);
        }).then(() => {
            expect(ag.baseElement).toMatchSnapshot();
        });
    });
});
