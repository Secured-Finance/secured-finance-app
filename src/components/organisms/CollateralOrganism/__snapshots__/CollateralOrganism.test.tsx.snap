// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollateralOrganism Component should render CollateralOrganism 1`] = `
<body>
  <div>
    <div
      class="h-[470px] w-[746px]"
    >
      <div
        class="h-full w-full overflow-hidden rounded-br-2xl rounded-bl-2xl border border-white-10 bg-[rgba(41,45,63,0.6)] shadow-tab"
      >
        <div
          aria-orientation="horizontal"
          class="flex h-[60px] border-b border-white-10"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="h-full focus:outline-none"
            id="headlessui-tabs-tab-1"
            role="tab"
            tabindex="0"
            type="button"
          >
            <div
              class="flex h-full w-full flex-col p-0 text-center"
            >
              <div
                class="h-1 w-full bg-starBlue"
              />
              <div
                class="flex h-full items-center justify-center px-8 bg-gradient-to-b from-tabGradient2 to-tabGradient1"
              >
                <button
                  class="typography-nav-menu-default h-4 text-neutral-8"
                  data-testid="Collateral Management-tab-button"
                >
                  Collateral Management
                </button>
              </div>
            </div>
          </button>
          <button
            aria-selected="false"
            class="h-full focus:outline-none"
            disabled=""
            id="headlessui-tabs-tab-2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <div
              class="flex h-full w-full flex-col p-0 text-center"
            >
              <div
                class="h-1 w-full "
              />
              <div
                class="flex h-full items-center justify-center px-8 "
              >
                <button
                  class="typography-nav-menu-default h-4 text-neutral-8"
                  data-testid="Scenario Analysis-tab-button"
                >
                  Scenario Analysis
                </button>
              </div>
            </div>
          </button>
        </div>
        <div>
          <div
            class="h-[410px] w-full"
          >
            <div
              class="flex h-full flex-row"
            >
              <div
                class="flex flex-col border-r border-white-10"
              >
                <div
                  class="h-80 w-64 border-b border-white-10"
                >
                  <div
                    class="m-6 flex flex-col gap-1"
                  >
                    <span
                      class="typography-body-2 h-6 w-fit text-slateGray"
                    >
                      Collateral Balance
                    </span>
                    <span
                      class="typography-big-body-bold w-fit text-white"
                    >
                      $0.00
                    </span>
                  </div>
                  <div
                    class="typography-caption ml-5 mt-8 w-40 text-grayScale"
                  >
                    Connect your wallet to see your deposited collateral balance.
                  </div>
                </div>
                <div
                  class="flex h-full flex-1 flex-row items-center justify-center gap-4"
                >
                  <button
                    aria-label="Deposit"
                    class="flex items-center justify-center rounded-xl bg-starBlue   undefined enabled:hover:bg-gradient-to-t enabled:hover:from-black-20 enabled:hover:via-black-20 enabled:hover:to-starBlue  disabled:bg-opacity-50 h-10 px-4 py-3 w-fit"
                    data-testid="deposit-collateral"
                    disabled=""
                  >
                    <p
                      class="text-white typography-button-2"
                    >
                      Deposit
                    </p>
                  </button>
                  <button
                    aria-label="Withdraw"
                    class="flex items-center justify-center rounded-xl bg-starBlue   undefined enabled:hover:bg-gradient-to-t enabled:hover:from-black-20 enabled:hover:via-black-20 enabled:hover:to-starBlue  disabled:bg-opacity-50 h-10 px-4 py-3 w-fit"
                    data-testid="withdraw-collateral"
                    disabled=""
                  >
                    <p
                      class="text-white typography-button-2"
                    >
                      Withdraw
                    </p>
                  </button>
                </div>
              </div>
            </div>
            <div
              style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
            />
            <div
              style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    id="headlessui-portal-root"
  >
    <div
      data-headlessui-portal=""
    >
      <div />
    </div>
    <div
      data-headlessui-portal=""
    >
      <div />
    </div>
  </div>
</body>
`;
